packages/block-editor/src/hooks/dimensions.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * WordPress dependencies
+ */
+import { hasBlockSupport } from '@wordpress/blocks';
+import { PanelBody } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { cleanEmptyObject } from './utils';
+import InspectorControls from '../components/inspector-controls';
+import useEditorFeature from '../components/use-editor-feature';
+import WidthControl from '../components/width-control';
+
+/**
+ * Key within block settings' supports array indicating support for
+ * width, e.g. settings found in 'block.json'.
+ */
+export const DIMENSIONS_SUPPORT_KEY = '__experimentalDimensions';
+
+export function DimensionsPanel( props ) {
+	const {
+		attributes: { style },
+		setAttributes,
+	} = props;
+
+	const widthOptions = useEditorFeature( 'dimensions.width' );
+	const isEnabled = useIsWidthEnabled( props );
+
+	if ( ! isEnabled ) {
+		return null;
+	}
+
+	const selectedWidth = getWidthFromAttributeValue(
+		widthOptions,
+		style?.dimensions?.width
+	);
+
+	function onChange( newWidth ) {
+		setAttributes( {
+			style: cleanEmptyObject( {
+				...style,
+				dimensions: {
+					...style?.dimensions,
+					width: newWidth,
+				},
+			} ),
+		} );
+	}
+
+	return (
+		<InspectorControls>
+			<PanelBody title={ __( 'Width Settings' ) }>
```

Comment:
To match with the Typography tools this could be "Dimensions" with "Width" becoming a label for the width controls within that Dimensions panel.

```suggestion
			<PanelBody title={ __( 'Dimensions' ) }>
```